File: src/medrelay/enums/incident_status.py
from enum import StrEnum


class IncidentStatus(StrEnum):
    NEW = "NEW"
    DISPATCHED = "DISPATCHED"
    EN_ROUTE = "EN_ROUTE"
    ON_SCENE = "ON_SCENE"
    TRANSPORTING = "TRANSPORTING"
    COMPLETE = "COMPLETE"
    CANCELLED = "CANCELLED"

    def label(self) -> str:
        return {
            IncidentStatus.NEW: "New",
            IncidentStatus.DISPATCHED: "Dispatched",
            IncidentStatus.EN_ROUTE: "En Route",
            IncidentStatus.ON_SCENE: "On Scene",
            IncidentStatus.TRANSPORTING: "Transporting",
            IncidentStatus.COMPLETE: "Complete",
            IncidentStatus.CANCELLED: "Cancelled",
        }[self]

File: app/Enums/IncidentStatus.php
<?php

declare(strict_types=1);

namespace App\Enums;

enum IncidentStatus: string
{
    case Dispatched = 'dispatched';
    case EnRoute = 'en_route';
    case OnScene = 'on_scene';
    case Transporting = 'transporting';
    case Complete = 'complete';
    case StoodDown = 'stood_down';

    /**
     * @return list<string>
     */
    public static function values(): array
    {
        return array_map(
            static fn (self $status): string => $status->value,
            self::cases(),
        );
    }

    public function label(): string
    {
        return match ($this) {
            self::Dispatched => 'Dispatched',
            self::EnRoute => 'En Route',
            self::OnScene => 'On Scene',
            self::Transporting => 'Transporting',
            self::Complete => 'Complete',
            self::StoodDown => 'Stood Down',
        };
    }
}
